### PR TITLE
Added delete route for events

### PR DESCRIPTION
**Pull request by abhinit75**
- BE completed
- Added UI for event deletion
- Confirm model included

Screenshot 2024-03-21 at 3 24 24 PM

Screenshot 2024-03-21 at 3 24 30 PM

### CODE REVIEW

File: server/src/api/EventRoutes.ts [hunk 1]
```diff
@@ -207,4 +207,41 @@ eventRoutes.get<void, MyEventResponse>(
   }
 );
 
+eventRoutes.delete("/delete/:eventId", async (req, res) => {
```
Comment (Kggupta) [inline server/src/api/EventRoutes.ts:210]: isClubAdmin middleware, remove the redundant logic you have checking if the user is club admin

File: server/src/api/EventRoutes.ts [hunk 2]
```diff
@@ -207,4 +207,41 @@ eventRoutes.get<void, MyEventResponse>(
...
+  const eventId = parseInt(req.params.eventId);
+  const userId = req.body.user.id;
+
+  try {
+    const event = await prisma.event.findUnique({
```
Comment (Kggupta) [inline server/src/api/EventRoutes.ts:215]: just delete and fail silently, this endpoint should be like 10 lines max.

File: server/src/api/EventRoutes.ts [hunk 3]
```diff
@@ -207,4 +207,41 @@ eventRoutes.get<void, MyEventResponse>(
...
+      return res
+        .status(403)
+        .send("User is not authorized to delete this event");
+    }
+
+    await prisma.event.delete({
+      where: { id: eventId },
+    });
+
+    res.status(200).send("Event deleted successfully");
```
Comment (Kggupta) [inline server/src/api/EventRoutes.ts:240]: not needed

File: server/src/api/EventRoutes.ts [hunk 4]
```diff
@@ -207,4 +207,41 @@ eventRoutes.get<void, MyEventResponse>(
...
+  } catch (error) {
+    console.error("Failed to delete event:", error);
```
Comment (Kggupta) [inline server/src/api/EventRoutes.ts:242]: not needed

File: server/src/api/EventRoutes.ts (hunk 1 above)
Comment (Kggupta) [reply server/src/api/EventRoutes.ts:210]: include the FE with this PR. you don't have to split prs by frontend and backend

**Review by Kggupta — APPROVED** [review]
comments

**Review by Kggupta — CHANGES_REQUESTED** [review]
comments

**Comment by Kggupta** [comment]
since nothing in this PR currently changes anything, it can be closed. or you can keep it open and add the FE changes in teh same pr

File: app/app/src/main/java/com/example/clubwat/viewmodels/EventDetailsViewModel.kt [hunk 5]
```diff
@@ -203,4 +203,26 @@ class EventDetailsViewModel(private val userRepository: UserRepository) : ViewMo
             }
         }
     }
+
+    fun deleteEvent(eventID: String) {
+        val id: Int = eventID.toInt()
+        viewModelScope.launch(Dispatchers.IO) {
+            try {
+                val obj = URL(BuildConfig.GET_EVENT_URL + "/delete" + "/$id")
```
Comment (Kggupta) [inline app/app/src/main/java/com/example/clubwat/viewmodels/EventDetailsViewModel.kt:211]: double check that this is the right path. idt it is

**Review by Kggupta — CHANGES_REQUESTED** [review]
comments

**Comment by abhinit75** [comment]
Just address comments by Keshav, otherwise lgtm!

File: app/app/src/main/java/com/example/clubwat/views/EventDetailsView.kt [hunk 6]
```diff
@@ -108,12 +112,25 @@ fun EventDetailsView(
                     )
                 },
                 actions = {
-                    IconButton(onClick = { navController.navigate("event/${eventId}/eventDetails") }) {
-                        Icon(
-                            imageVector = Icons.Default.Edit,
-                            contentDescription = "Edit"
-                        )
+                    profile.value?.let { profileValue ->
+                        if (profileValue.adminFlag) {
```
Comment (Kggupta) [inline app/app/src/main/java/com/example/clubwat/views/EventDetailsView.kt:116]: this is superadmin, i'll fix this one sec

**Review by Kggupta — APPROVED** [review]

**Review by mary1afshar — APPROVED** [review]

**Pull request merged**